notification/slack/channel.go

```diff
@@ -278,8 +278,16 @@ func alertMsgOption(ctx context.Context, callbackID string, id int, summary, det
 		details = ""
 	}
 	if details != "" {
+		escaped := slackutilsx.EscapeMessage(details)
+		if len(escaped) > 3000 {
+			newDetailsLen := len(details) - (len(escaped) - 3000)
+			if newDetailsLen < 0 {
+				newDetailsLen = 0
+			}
+			escaped = slackutilsx.EscapeMessage(details[:newDetailsLen])
```

Comment:
Yeah, it's not perfect -- best effort really. It's the same logic we apply to SMS messages -- we trim the input by the amount the output is off. But now that you mention it, maybe I'll do some quick testing to see if it can be abused to still break -- unlike SMS the output in this case will always be larger than the input.

Reply:
^ same or larger